### ISSUE TITLE
10 languages = 30-50 navigation menu blocks

### ISSUE THREAD

**Issue by bd0bd** [backdrop/backdrop-issues#1690]
10 main menu blocks
10 services menu blocks
10 about menu blocks

It is minimum 30 menu blocks we need for a small multi lang website.
If we add additional ones then it will be 40 or 50 menu blocks.

Also we would need to have additional 10 header blocks for each language.

It is too much blocks.
I think it is sickly abnormal situation for a respectable CMS like Backdrop.
We are respectable I think so :)

This CMS is an international one and it is not a new on the market but still it has no a normal and easy solution implemented.

**Comment by findlabnet** [comment]
@bd0bd, you absolutely right: full i18n implementation - a must in today's world. Even for Drupal 6 site by using this contribute 10-15 languages is not a problem.

**Comment by herbdool** [comment]
Backdrop CMS, since it is born out of Drupal 7, currently only has whatever Drupal 7 had for multilingual support in core. If you've ever built a multilingual website with Drupal 7 you'll know that for complex multilingual support you soon have to use the [Internationalization](http://drupal.org/project/i18n) (or i18n) and/or [Entity Translation](http://drupal.org/project/entity_translation) contrib modules.

I currently don't need i18n but figured I would eventually so recently I attempted to [port Internationalization to Backdrop](https://github.com/herbdool/i18n) and made some progress with the base module. But that module is a complex beast and it would take some time to port all the submodules to Backdrop. One key issue is that i18n relies on the Drupal Variable module which is based on the old configuration in the database system in Drupal and not the new configuration management system in Backdrop (or Drupal 8). And even if these modules get ported, the core maintainers may go a different route whenever they are able to provide more multilingual support in core.

In short, porting this could use other people's help. I'm not working on it at the moment (but may pick it up again at some point in the future).

**Comment by bd0bd** [comment]
@findlabnet, do you mean this module is a complete solution (not only for menu blocks but for all other aspects) for multi lag website?

**Comment by bd0bd** [comment]
@herbdool, of cause it would be very good if somebody would port it.

**Comment by findlabnet** [comment]
@bd0bd 
>this module is a complete solution

silver bullet doesn't exists at all, but for example, one menu with 10 translations seems better than 10 menus - one for each language. Anyway, features of this module very good documented in Drupal world.

**Comment by herbdool** [comment]
There's been a start at porting i18n but still a ways to go.

**Comment by bd0bd** [comment]
I know. I have already try it.

**Comment by herbdool** [comment]
I haven't tried it but I doubt it's usable yet. I started porting it separately and found it required a lot of work.

**Comment by bd0bd** [comment]
4  main problems with multi lang site:

menu
home page
header
views pages

These are the key things should be solved first in my opinion.

**Comment by BWPanda** [comment]
Going to close this in favour of something like https://github.com/backdrop/backdrop-issues/issues/3369 where different issues can be separated and and worked on independently.
